data/topic-schema/src/main/resources/net/corda/schema/CPI.yaml

```diff
@@ -0,0 +1,22 @@
+topics:
+  CPIInfoTopic:
+    name: cpi.info
+    consumers:
+    producers:
+    config:
+      cleanup.policy: compact
+      segment.ms: 300000
+      delete.retention.ms: 300000
+      min.compaction.lag.ms: 60000
+      max.compaction.lag.ms: 300000
+      min.cleanable.dirty.ratio: 0.5

```

Comment:
I don't know enough about the topic usage to answer this question but, do we really want/need the ability to tweak the configuration per individual topic vs having different classes of topic? I also don't know whether it's a problem that this is obviously very Kafka specific. Having different classes of topic would make it easier to have a separation between the generic topic definitions and their specific rendering for a given message bus.

Reply:
I would like to have classes too, with the option to add the odd override. I don't think Ben was sure how best to approach that though

Reply:
Spoke to Matthew about this, big issue was that jackson didn't support anchors. so some more work will be needed to figure out how to do this without breaking anything or making the toolchain more complicated

Reply:
I wasn't expecting anything nearly as clever as the use of anchors - just the name of a class/type here and another file that defined the properties for that class/type. It can certainly wait though.